src/voxelalgorithms.cpp

```diff
@@ -1263,44 +1263,54 @@ void repair_block_light(ServerMap *map, MapBlock *block,
 		modified_blocks);
 }
 
-VoxelLineIterator::VoxelLineIterator(const v3f &start_position, const v3f &line_vector) :
-	m_start_position(start_position),
-	m_line_vector(line_vector)
+v3f shrink_to_map_limit(v3f pos) {
+	const float limit = MAX_MAP_GENERATION_LIMIT;
+	pos.X = std::isnormal(pos.X) ? std::max(std::min(pos.X, limit), -limit) : 0;
+	pos.Y = std::isnormal(pos.Y) ? std::max(std::min(pos.Y, limit), -limit) : 0;
+	pos.Z = std::isnormal(pos.Z) ? std::max(std::min(pos.Z, limit), -limit) : 0;
+	return pos;
+}
+
+VoxelLineIterator::VoxelLineIterator(const v3f &start_position, const v3f &end_position) :
+	m_start_position(shrink_to_map_limit(start_position)),
```

Comment:
Welp that approach doesn't work. Raycast vector would have a wrong angle. Guess I will accurately adopt SmallJoker's approach.